src/extension/op_def.rs

```diff
@@ -326,36 +343,40 @@ impl Extension {
         &mut self,
         name: SmolStr,
         description: String,
-        params: Vec<TypeParam>,
+        params: (Vec<TypeParam>, Vec<TypeParam>),
         misc: HashMap<String, serde_yaml::Value>,
         lower_funcs: Vec<LowerFunc>,
         signature_func: impl CustomSignatureFunc + 'static,
     ) -> Result<&OpDef, ExtensionBuildError> {
+        let mut all_params = params.0;
+        let num_static_params = all_params.len();
+        all_params.extend(params.1);
         self.add_op(
             name,
             description,
             misc,
             lower_funcs,
             SignatureFunc::CustomFunc {
-                params,
+                all_params,
+                num_static_params,
                 func: Box::new(signature_func),
             },
         )
     }
 
-    /// Create an OpDef with custom binary code to compute the signature, and no "misc" or "lowering
-    /// functions" defined.
+    /// Create an OpDef with custom binary code to compute the signature,
+    /// which is not polymorphic; and no "misc" or "lowering functions" defined.
```

Comment:
we might well change this when we've converted the existing uses in the extensions to use type schemes rather than their own custom binary functions

Reply:
Maybe add a TODO?